src/ParameterLists.f90

```diff
@@ -4718,12 +4718,14 @@ SUBROUTINE edit_ParamType_STR(thisParam,funit,indent,prefix,paddtw)
         ENDIF
       ENDIF
       WRITE(fmt,'(i12)') i; fmt=ADJUSTL(fmt)
+      tmpstr=''
+      IF(LEN_TRIM(thisParam%val) > 0) tmpstr=thisParam%val
```

Comment:
I think there are some other parameter types for arrays of strings that also need to get modified.